
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>2. gComposite.py &#8212; epage 1.0.0 documentation</title>
    <link rel="stylesheet" href="../_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="../_static/language_data.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Version 1.0.0" href="../change_log.html" />
    <link rel="prev" title="1. SVM_performance.py" href="SVM_performance.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="gcomposite-py">
<h1>2. gComposite.py<a class="headerlink" href="#gcomposite-py" title="Permalink to this headline">¶</a></h1>
<div class="section" id="description">
<h2>2.1. Description<a class="headerlink" href="#description" title="Permalink to this headline">¶</a></h2>
<p>This program Calculates these Composite Expression Scores. Compared to expression score of a single gene,
<em>composite expression score</em> measure the overall activity of <strong>a set of genes</strong>. It is often used
to measure the activity of a pathway or transcription factor.</p>
<p>It calculates these scores:</p>
<blockquote>
<div><ul class="simple">
<li><p>Gene Set Variation Analysis (GSVA). <a class="footnote-reference brackets" href="#id5" id="id1">1</a></p></li>
<li><p>Single Sample GSEA (ssGSEA). <a class="footnote-reference brackets" href="#id6" id="id2">2</a></p></li>
<li><p>zscore <a class="footnote-reference brackets" href="#id7" id="id3">3</a></p></li>
<li><p>plage <a class="footnote-reference brackets" href="#id8" id="id4">4</a></p></li>
</ul>
</div></blockquote>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The R package <a class="reference external" href="https://bioconductor.org/packages/release/bioc/html/GSVA.html">GSVA</a> will be automatically installed
and used to calculate these scores.</p>
</div>
</div>
<div class="section" id="options">
<h2>2.2. Options<a class="headerlink" href="#options" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><dl class="option-list">
<dt><kbd><span class="option">--version</span></kbd></dt>
<dd><p>show program’s version number and exit</p>
</dd>
<dt><kbd><span class="option">-h</span>, <span class="option">--help</span></kbd></dt>
<dd><p>show this help message and exit</p>
</dd>
<dt><kbd><span class="option">-e <var>EXPR_FILE</var></span>, <span class="option">--expr_matrix=<var>EXPR_FILE</var></span></kbd></dt>
<dd><p>Tab-separated data matrix file containing gene
expression values. The 1st row containing
sample/patient IDs and the 1st column containing gene
symbols(mut be unique). File can be compressed (.gz,
.Z, .z, .bz, .bz2, bzip2).</p>
</dd>
<dt><kbd><span class="option">-g <var>GENE_FILE</var></span>, <span class="option">--gene=<var>GENE_FILE</var></span></kbd></dt>
<dd><p>GMT file. The GMT file format is a tab delimited file
format that describes gene sets (Each gene set is
described by a name, a description, and the genes in
the gene set). In the GMT format, each row represents
a gene set. The first column is get set name (must be
unique). The second column is brief description (can
be ‘na’).</p>
</dd>
<dt><kbd><span class="option">-k <var>GROUP_FILE</var></span>, <span class="option">--group=<var>GROUP_FILE</var></span></kbd></dt>
<dd><p>Group file (in CSV format). First column is sample ID,
second column is group ID</p>
</dd>
<dt><kbd><span class="option">-s <var>SAMPLE_FILE</var></span>, <span class="option">--sample=<var>SAMPLE_FILE</var></span></kbd></dt>
<dd><p>Sample list file containing sample IDs. Each row can
be a single sample ID, a comma-separated sample IDs or
a space-separated sample IDs. Sample IDs must match
exactly to those in the data matrix file. If omitted,
calculated activity scores for <em>all</em> the samples. File
can be compressed (.gz, .Z, .z, .bz, .bz2, bzip2).
default=none (All samples will be used)</p>
</dd>
<dt><kbd><span class="option">-l</span>, <span class="option">--log</span></kbd></dt>
<dd><p>If True, will do log2(x+1) transformation for gene
experssion values. Must set to ‘True’ if expressin
values are RNA-seq count. default=False</p>
</dd>
<dt><kbd><span class="option">-p <var>N_THREAD</var></span>, <span class="option">--processor=<var>N_THREAD</var></span></kbd></dt>
<dd><p>Number of processors to use when doing the
calculations in parallel. default=0 (use all available
processors)</p>
</dd>
<dt><kbd><span class="option">-o <var>OUT_FILE</var></span>, <span class="option">--output=<var>OUT_FILE</var></span></kbd></dt>
<dd><p>The prefix of the output file.</p>
</dd>
</dl>
</div></blockquote>
</div>
<div class="section" id="input-files-examples">
<h2>2.3. Input files (examples)<a class="headerlink" href="#input-files-examples" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Gene expression table. Example: lung_expr.81genes.tsv</p></li>
<li><p>Gene list in GMT format. Example: lung_p53_target.gmt</p></li>
<li><p>Group file. Example: lung_group.csv</p></li>
</ul>
</div>
<div class="section" id="command">
<h2>2.4. Command<a class="headerlink" href="#command" title="Permalink to this headline">¶</a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$  python3  gComposite.py  -e lung_expr.81genes.tsv -g  lung_p53_target.gmt  -k lung_group.csv -o lung
</pre></div>
</div>
</div>
<div class="section" id="output-files">
<h2>2.5. Output files<a class="headerlink" href="#output-files" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><ul class="simple">
<li><p>output.R : R script to run GSVA package</p></li>
<li><p>output.mat.tsv : Data that is actually used. Might be the same as the input “lung_expr.81genes.tsv”, or just a subset of “lung_expr.81genes.tsv”.</p></li>
<li><p>output_combined.csv : comma-separated composite expression score (group IDs were included)</p></li>
<li><p>output_combined.tsv : TAB-separated composite expression score (group IDs were NOT included)</p></li>
<li><p>output_gsva.csv : GSVA scores</p></li>
<li><p>output_pca.csv : First two principal components of PCA.</p></li>
<li><p>output_plage.csv : PLAGE scores</p></li>
<li><p>output_ssgsea.csv : ssGSEA scores</p></li>
<li><p>output_zscore.csv : Z-scores</p></li>
</ul>
</div></blockquote>
<p>The file “output_combined.csv” contains everything!</p>
</div>
<div class="section" id="references">
<h2>2.6. References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<dl class="footnote brackets">
<dt class="label" id="id5"><span class="brackets"><a class="fn-backref" href="#id1">1</a></span></dt>
<dd><p>Hänzelmann S, Castelo R, Guinney J. GSVA: gene set variation analysis for microarray and RNA-seq data. BMC Bioinformatics. 2013;14:7. Published 2013 Jan 16. doi:10.1186/1471-2105-14-7</p>
</dd>
<dt class="label" id="id6"><span class="brackets"><a class="fn-backref" href="#id2">2</a></span></dt>
<dd><p>Barbie DA, Tamayo P, Boehm JS, et al. Systematic RNA interference reveals that oncogenic KRAS-driven cancers require TBK1. Nature. 2009;462(7269):108-112. doi:10.1038/nature08460</p>
</dd>
<dt class="label" id="id7"><span class="brackets"><a class="fn-backref" href="#id3">3</a></span></dt>
<dd><p>Lee E, Chuang HY, Kim JW, Ideker T, Lee D. Inferring pathway activity toward precise disease classification. PLoS Comput Biol. 2008;4(11):e1000217. doi:10.1371/journal.pcbi.1000217</p>
</dd>
<dt class="label" id="id8"><span class="brackets"><a class="fn-backref" href="#id4">4</a></span></dt>
<dd><p>Tomfohr J, Lu J, Kepler TB. Pathway level analysis of gene expression using singular value decomposition. BMC Bioinformatics. 2005;6:225. Published 2005 Sep 12. doi:10.1186/1471-2105-6-225</p>
</dd>
</dl>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../index.html">epage</a></h1>








<h3>Navigation</h3>
<p class="caption"><span class="caption-text">Overview</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../overview.html">overview</a></li>
</ul>
<p class="caption"><span class="caption-text">Installation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Prerequisites</a></li>
<li class="toctree-l1"><a class="reference internal" href="../installation.html#python-dependencies">Python dependencies</a></li>
<li class="toctree-l1"><a class="reference internal" href="../installation.html#install">Install</a></li>
<li class="toctree-l1"><a class="reference internal" href="../installation.html#upgrade">Upgrade</a></li>
</ul>
<p class="caption"><span class="caption-text">Usage Information</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="SVM_performance.html">1. SVM_performance.py</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">2. gComposite.py</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#description">2.1. Description</a></li>
<li class="toctree-l2"><a class="reference internal" href="#options">2.2. Options</a></li>
<li class="toctree-l2"><a class="reference internal" href="#input-files-examples">2.3. Input files (examples)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#command">2.4. Command</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output-files">2.5. Output files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#references">2.6. References</a></li>
</ul>
</li>
</ul>
<p class="caption"><span class="caption-text">Release history</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../change_log.html">Version 1.0.0</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
      <li>Previous: <a href="SVM_performance.html" title="previous chapter">1. SVM_performance.py</a></li>
      <li>Next: <a href="../change_log.html" title="next chapter">Version 1.0.0</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2020, Liguo Wang, et al.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 2.1.2</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../_sources/demo/gComposite.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>